lib/fluent/plugin/buffer.rb

```diff
@@ -761,18 +771,41 @@ def write_step_by_step(metadata, data, format, splits_count, &block)
             begin
               while writing_splits_index < splits.size
                 split = splits[writing_splits_index]
+                formatted_split = format ? format.call(split) : split.first
+                if split.size == 1 && original_bytesize == 0
+                  if format == nil && @compress
```

Comment:
`@compress` is `:text` or `:gzip`

https://github.com/fluent/fluentd/blob/6d386249f3225ffa5794d5a45afe19c9adbbabb5/lib/fluent/plugin/buffer.rb#L67

Only `:gzip` is undetermined here.

Reply:
Thanks, I'll check it.

Reply:
Added check for `:gzip`